lms/djangoapps/courseware/models.py

```diff
@@ -260,3 +260,20 @@ class Meta:
 
     def __unicode__(self):
         return "[OCGLog] %s: %s" % (self.course_id.to_deprecated_string(), self.created)  # pylint: disable=no-member
+
+
+class StudentFieldOverride(models.Model):
+    """
+    Holds the value of a specific field overriden for a student.  This is used
+    by the code in the `courseware.student_field_overrides` module to provide
+    overrides of xblock fields on a per user basis.
+    """
+    course_id = CourseKeyField(max_length=255, db_index=True)
+    location = LocationKeyField(max_length=255, db_index=True)
+    student = models.ForeignKey(User, db_index=True)
+
+    class Meta:   # pylint: disable=missing-docstring
+        unique_together = (('course_id', 'location', 'student'),)
+
+    field = models.CharField(max_length=255)
+    value = models.TextField(default='null')
```

Comment:
I guess I haven't really thought through the full analytics implications. We have been operating under the assumption that we can reconstruct the "state" of the course content at any point in time from the versioned course content (provided by split). It sounds like we would need to augment that analysis with whatever student specific overrides were setup at the time.

Perhaps we could include a context field in the event that enumerates any field overrides that were present when the event was fired? Then we wouldn't need to maintain some sort of historical state table that would tell us what overrides were present at what times etc. For a problem_check event, for example, we could include whatever overrides apply to that student and that module.

Similar to the way we track A/B tests:

https://github.com/edx/edx-platform/blob/d4de932c2b46dbe1ad6439731b4312fb36813d6d/openedx/core/djangoapps/user_api/middleware.py